src/Npgsql/TypeHandlers/CompositeHandlers/CompositeMemberHandlerOfStruct.cs

```diff
@@ -68,7 +68,7 @@ public override async ValueTask Read(ByReference<TComposite> composite, NpgsqlRe
                 return;
 
             var value = NullableHandler<TMember>.Exists
-                ? await NullableHandler<TMember>.ReadAsync(_handler, buffer, length, async, null)
+                ? await NullableHandler<TMember>.ReadAsync!(_handler, buffer, length, async, null)
```

Comment:
Are you aware of the new pre- and post-condition nullability attributes ([see this doc](https://github.com/dotnet/csharplang/blob/master/meetings/2019/LDM-2019-05-15.md))? I am planning to do another pass on the code base to see where they can help, so you don't need to worry about it now.